### PR TITLE
Remove multithreading in HDR decoding

### PR DESCRIPTION
**Pull request by bash**
As discussed in https://github.com/image-rs/image/issues/772#issuecomment-1382406241. This also makes it work on WASM.

---

I license past and future contributions under the dual MIT/Apache-2.0 license,
allowing licensees to choose either at their option.

### CODE REVIEW

File: src/codecs/hdr/decoder.rs [hunk 1]
```diff
@@ -460,21 +459,16 @@ impl<R: BufRead> HdrDecoder<R> {
...
+        let mut buf = vec![Default::default(); self.width as usize];
+        for chunk in chunks_iter {
+            read_scanline(&mut self.r, &mut buf[..])?;
+            for (dst, &pix) in chunk.iter_mut().zip(buf.iter()) {
+                *dst = f(pix);
             }
-            Ok(())
-        }) as Result<(), ImageError>)?;
+            buf.clear();
+            buf.resize(self.width as usize, Default::default());
```
Comment (fintelia) [inline src/codecs/hdr/decoder.rs:470]: Would `buf.fill(Default::default())` work instead?

Comment (fintelia) [reply src/codecs/hdr/decoder.rs:470]: Actually, why is this needed at all? Shouldn't read_scanline overwrite the entire buffer?

**Comment by fintelia** [comment]
The same CI failure seems to be happening on main: https://github.com/image-rs/image/actions/runs/4067817240/jobs/7156440418

File: src/codecs/hdr/decoder.rs (hunk 1 above)
Comment (bash) [reply src/codecs/hdr/decoder.rs]: Removed (and commented) in [7df5cb1](https://github.com/image-rs/image/pull/1857/commits/7df5cb142318525ff1c544361656157c6bb70be6)

**Review by fintelia — APPROVED** [review]

**Pull request merged**